spanner/session.go

```diff
@@ -832,10 +832,13 @@ func (p *sessionPool) errGetSessionTimeoutWithTrackedSessionHandles(code codes.C
 // requires the caller to have locked p.mu.
 func (p *sessionPool) getTrackedSessionHandleStacksLocked() string {
 	p.mu.Lock()
-	defer p.mu.Unlock()
+	element := p.trackedSessionHandles.Front()
+	// NOTE: No need to keep the sessionPool lock since we don't modify it,
+	// this will return the snapshot of current elements when error getting sessions
```

Comment:
`trackedSessionHandles` could be being modified in different places like here. https://github.com/googleapis/google-cloud-go/blob/spanner/v1.30.0/spanner/session.go#L83

How do we guarantee that we will dump a consistent snapshot of `trackedSessionHandles` without locking `p.mu`?

Reply:
I think getTrackedSessionHandleStacksLocked is only called when the client gets `errGetSessionTimeout`, so even if trackedSessionHandles later gets updated as a customer I want to know what happen when errGetSessionTimeout was thrown and not the current snapshot, so I think we should release the lock as soon as we read trackedSessionHandles.Front()

Reply:
@codyoss can you please help review this.

Reply:
@rahul2393 If I understand correctly, this could print invalid state as stack traces.

Let's say there are the following session handle chains in `trackedSessionHandles` when the client gets `errGetSessionTimeout`. And same time, let's say `shA` was [recycled](https://github.com/googleapis/google-cloud-go/blob/spanner/v1.30.0/spanner/session.go#L83) by other goroutine.

```
shA -> shB -> shC
```

Then valid state would be either `shA -> shB -> shC` or `shB -> shC` depending on timinig.

But if we allow other goroutine to get `shA` removed from `trackedSessionHandles`, then the returned value from `p.trackedSessionHandles.Front()` would loose `next` pointer as Go's [list.List.Remove](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/container/list/list.go;l=108-115;drc=refs%2Ftags%2Fgo1.18) removes the next pointer from the element. So the stack trace would be just `shA` for this case.

Reply:
you are right, I updated the PR @yfuruyama can you check if its ok